### PR TITLE
Added comments to source file - lpae.c, lpae.h, smp.h

### PR DESCRIPTION
**Pull request by xiane**
Add comments to lpae.c, lpae.h, smp.h

### CODE REVIEW

**Comment by kesl** [comment]
Can one of the admins verify this patch?

File: hypervisor/hardware/arm32ve/libhw/lpae.c [hunk 1]
```diff
@@ -3,13 +3,27 @@
 #include <log/print.h>
 #include <log/uart_print.h>
 
-/* Long-descriptor translation table format */
+/**
+ * \defgroup LPAE_address_mask
+ *
+ * The address mask of each level descriptors.
+ * - TTBL_L1_OUTADDR_MASK[39:30] Level1 1GB Block address mask.
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.c:10]: According to your description, the names of macro such as TTBL_L1_OUTADDR_MASK and TTBL_L1_TABADDR_MASK are not good naming now. What do you think about it?

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 2]
```diff
@@ -2,21 +2,223 @@
 #define __LPAE_H__
 #include <arch_types.h>
 
+/**
+ * @file lpae.h
+ *
+ * <pre>
+ * ARMv7-A LPAE pagetables:3-level trie, mapping 40-bit input to
+ * 40-bit output addresses.  Tables at all levels have 512 64-bit entries
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:10]: Remove additional space after "addresses.".
s/addresses.  Tables/addresses. Tables

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 3]
```diff
@@ -2,21 +2,223 @@
...
+ * (i.e. are 4Kb long).
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:11]: what does 'are 4Kb long' mean?

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 4]
```diff
@@ -2,21 +2,223 @@
...
+ * </pre>
+ */
+
+/**
+ * \defgroup Long-descriptor
+ *
+ * @brief Long-descriptor_translation_table_format
+ *
+ * Long-descriptor translation table format supports the assignment of memory
+ * attributes to memory Pages, at a granularity of 4KB, across the complete
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:26]: s/Pages/pages

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 5]
```diff
@@ -2,21 +2,223 @@
...
+ * - a block entry, that defines the memory properties for the access.
+ * - a reserved format.
+ *
+ * Bit[1] of the descriptor indicates the descriptor type, and bit[0] indicates
+ * whether the descriptor is valid.
+ *
+ * - First-level and second-level descriptor formats
+ *  - block descriptor
+ * <pre>
+ * |bit  |0|1|2        11|12   n-1|n           39|40    51|52       63|
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:52]: Confused, Why did you write different bit ordering between
'|bit  |0|1|2        11|12   n-1|n           39|40    51|52       63|' and
|     | | |attributes |        |[39:n]        |        |attributes |.

File: hypervisor/include/smp.h [hunk 6]
```diff
@@ -6,12 +6,17 @@
...
 #define MPIDR_MASK 0xFFFFFF
 
+/**
+ * @brief Return the smp processor id
+ *
+ * Read the processor id from Multiprocessor ID Register(MPIDR) and return it.
+ * - Coretex-A15
+ *   - MPIDR[1:0] - CPUID - 0, 1, 2, OR 3
+ *   - MPIDR[7:2] - Rsereved, Read as zero
+ * @return Smp processor id.
```
Comment (Wonseok) [inline hypervisor/include/smp.h:16]: Wrong description.

File: hypervisor/include/smp.h [hunk 7]
```diff
@@ -6,12 +6,17 @@
 
 #define MPIDR_MASK 0xFFFFFF
 
+/**
+ * @brief Return the smp processor id
+ *
+ * Read the processor id from Multiprocessor ID Register(MPIDR) and return it.
+ * - Coretex-A15
+ *   - MPIDR[1:0] - CPUID - 0, 1, 2, OR 3
+ *   - MPIDR[7:2] - Rsereved, Read as zero
```
Comment (Wonseok) [inline hypervisor/include/smp.h:15]: s/Rsereved/Reserved

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 8]
```diff
@@ -2,21 +2,223 @@
...
+ * - is used for
+ *  - the Non-secure PL2 stage-1 translation.
+ *  - the Non-secure PL1&0 stage-2 translation.
+ * - can be used for the Secure and Non-secure PL1&0 stage-1 translations.
+ *
+ * the Long-descriptor translation table format provides up to three levels
+ * of address lookup. A translation table wlak starts either at the first level
+ * or the second level of address lookup.
+ *
+ * In general, a descriptor is one of
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:40]: I have a question. How do we use the descriptor?
We don't want to learn all of LPAE, we just want to know what we used?

Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:40]: Another question. If we use one of them and then the one is optimal way to use LPAE for us?

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 9]
```diff
@@ -2,21 +2,223 @@
...
+ * input address range. It also supports the assignment of memory attributes to
+ * blocks of memory, where a block can be 2MB or 1GB.
+ *
+ * This foramt
+ * - is used for
+ *  - the Non-secure PL2 stage-1 translation.
+ *  - the Non-secure PL1&0 stage-2 translation.
+ * - can be used for the Secure and Non-secure PL1&0 stage-1 translations.
+ *
+ * the Long-descriptor translation table format provides up to three levels
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:36]: s/the Long-descriptor/The long descriptor

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 10]
```diff
@@ -2,21 +2,223 @@
...
+ * Long-descriptor translation table format supports the assignment of memory
+ * attributes to memory Pages, at a granularity of 4KB, across the complete
+ * input address range. It also supports the assignment of memory attributes to
+ * blocks of memory, where a block can be 2MB or 1GB.
+ *
+ * This foramt
+ * - is used for
+ *  - the Non-secure PL2 stage-1 translation.
+ *  - the Non-secure PL1&0 stage-2 translation.
+ * - can be used for the Secure and Non-secure PL1&0 stage-1 translations.
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:34]: Odd indentation L31-L34.

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 11]
```diff
@@ -2,21 +2,223 @@
...
+ *
+ * @brief Long-descriptor_translation_table_format
+ *
+ * Long-descriptor translation table format supports the assignment of memory
+ * attributes to memory Pages, at a granularity of 4KB, across the complete
+ * input address range. It also supports the assignment of memory attributes to
+ * blocks of memory, where a block can be 2MB or 1GB.
+ *
+ * This foramt
+ * - is used for
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:31]: I think this comments want to locate after string "the format", isn't it?

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 12]
```diff
@@ -2,21 +2,223 @@
...
+ * - an invalid or fault entry.
+ * - a table entry, that points the next-level translation table.
+ * - a block entry, that defines the memory properties for the access.
+ * - a reserved format.
+ *
+ * Bit[1] of the descriptor indicates the descriptor type, and bit[0] indicates
+ * whether the descriptor is valid.
+ *
+ * - First-level and second-level descriptor formats
+ *  - block descriptor
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:50]: Odd indentation.

File: hypervisor/hardware/arm32ve/libhw/lpae.h [hunk 13]
```diff
@@ -2,21 +2,223 @@
...
+ * </pre>
+ *  - The first-level descriptor returns the address of the second-level table.
+ *  - The second-level descriptor returns the address of the third-level talbe.
+ *
+ * - Third-level descriptor formats
+ *  - page descriptor
+ * <pre>
+ * |bits |0|1|2        11|12          39|40    51|52       63|
+ * |value|1|1|Lower block|Output address|UNK/SBZP|Upper block|
+ * |     | | |attributes |[39:12]       |        |attributes |
```
Comment (Wonseok) [inline hypervisor/hardware/arm32ve/libhw/lpae.h:72]: What does this line mean?

**Comment by Wonseok** [comment]
Overall looks good to me, but you might want to fix some comments.
1. Please distinguish singular and plural.
2. Please use period(.) at the end of each sentence.
3. Please do not use dash(-) in a sentence such as Long-descriptor_translation_table_format.
4. Could you please summarize it more? It's too much detail. See the Long-descriptor parts.

**Pull request closed without merging**